layer/contracts/logged_data.py

```diff
@@ -395,3 +398,92 @@ class Markdown:
     """
 
     data: str
+
+
+class LoggedDataObject:
+    def __init__(self, logged_data: LoggedData, epoch: Optional[int] = None):
+        self._logged_data = logged_data
+        self._epoch = epoch
+
+    @staticmethod
+    def _download_object_to(url: str, file_path: Path) -> None:
+        response = requests.get(url)
+        with open(file_path, "wb") as filer_handler:
+            filer_handler.write(response.content)
+
+    def value(self) -> Union[str, float, bool, pd.DataFrame, "PIL.Image.Image"]:
+        """
+        To be called for string, numeric, boolean, image, markdown and table values.
+        """
+        if self.is_table():
+            return pd.read_json(self._logged_data.data, orient="table")
+        elif self.is_number():
+            return float(self._logged_data.data)
+        elif self.is_text():
+            return self._logged_data.data
+        elif self.is_boolean():
+            return bool(self._logged_data.data)
+        elif self.is_image():
+            from PIL.Image import open as pil_open
+
+            url = (
+                self._logged_data.data
+                if self._epoch is None
+                else self._logged_data.epoched_data[self._epoch]
+            )
+            with tempfile.NamedTemporaryFile() as temp_file:
+                self._download_object_to(url, Path(temp_file.name))
+                image = pil_open(temp_file)
+                image.load()
+                return image
+        elif self.is_markdown():
+            return self._logged_data.data
+        else:
+            raise Exception(
```

Comment:
Maybe have a test for this behavior (same for `download_to`'s such exception too)?